rust/main/chains/hyperlane-cosmos/src/providers/cosmos/provider.rs

```diff
@@ -103,6 +110,46 @@ impl CosmosProvider {
         Ok((account_id, signer_info.sequence))
     }
 
+    fn normalize_public_key(
+        &self,
+        signer_public_key: SignerPublicKey,
+    ) -> ChainResult<SignerPublicKey> {
+        let public_key = match signer_public_key {
+            SignerPublicKey::Single(pk) => SignerPublicKey::from(pk),
+            SignerPublicKey::LegacyAminoMultisig(pk) => SignerPublicKey::from(pk),
+            SignerPublicKey::Any(pk) => {
+                if pk.type_url != INJECTIVE_PUBLIC_KEY_TYPE_URL {
+                    let msg = "we don't support this kind of TYPE_URL for public key";
+                    warn!(pk.type_url, msg);
+                    Err(HyperlaneCosmosError::PublicKeyError(msg.to_owned()))?
+                }
+
+                let any = Any {
+                    type_url: PublicKey::SECP256K1_TYPE_URL.to_owned(),
+                    value: pk.value,
+                };
+
+                let proto = proto::cosmos::crypto::secp256k1::PubKey::from_any(&any)
+                    .map_err(HyperlaneCosmosError::Prost)?;
```

Comment:
can avoid hardcoding the error to map to by doing `.map_err(Into::<HyperlaneCosmosError>::into)?`

Reply:
Fixed